### ISSUE TITLE
Mismatch between this package and gitignore files

### ISSUE THREAD

**Issue by arcanis** [kaelzhang/node-ignore#56]
Given the following reproduction:

```bash
mkdir /tmp/ignoretest
cd /tmp/ignoretest

mkdir foo
touch foo/bar.js

echo > .gitignore <<'EOT'
/*/
!/foo/
EOT

git init
git add .
git status
```

The `foo/bar.js` file got added to the repository. Doing the same on `ignore`, however:

```js
const ignore = require(`ignore`);

ignore().add([
  `/*/`,
  `!/foo/`,
]).ignores(`foo/bar.js`) // true
```

**Comment by kaelzhang** [comment]
gitignore rules changes in [2.22.1](https://git-scm.com/docs/gitignore/2.22.1)!

**Comment by kaelzhang** [comment]
landed in [`5.1.3`](https://github.com/kaelzhang/node-ignore/releases/tag/5.1.3)
